index.js

```diff
@@ -0,0 +1,151 @@
+const axios = require("axios"); // A small REST client
+const isemail = require("isemail"); // Email address check
+
+function handlePOST(req, res) {
+  // We need query strings to set some parameters
+  const query = req.query;
+  const body = req.body;
+
+  // Let's get user input here from the ChatFuel request
+  const firstName = body["first name"] || "Firstname",
+    lastName = body["last name"] || "Lastname",
+    formResponse = body.formResponse || "No response",
+    messengerUserId = body["messenger user id"] || "none-provided",
+    preferAnonymous = body.PreferAnonymous || "Yes",
+    location = body.location || "none-provided";
+
+  // Some initial variables
+  let isAnon = true, // Default to anonymous
+    sourceId = 361, // use ?sourceId=361 in Chatfuel to change this
+    email = body.email || "noemail",
+    hearkenName = query.hearkenName || "abc"; // use hearkenName=
+
+  // Make sourceId query string a number and provide fallback
+  sourceId = +query.sourceId || 361; // Nothing special about 361 just the orig number
+
+  // Set the Hearken endpoint. This is exposed through the Hearken embeds
+  // https://modules.wearehearken.com/abc/embed/361/share
+  const hearkenUrl =
+    "https://modules.wearehearken.com/" +
+    encodeURIComponent(hearkenName) +
+    "/api/questions.js";
+
+  // Hearken accepts a Full Name field
+  const fullName = firstName + " " + lastName;
+
+  // Find out if they want to remain anon or not
+  // TODO: Do some actual language processing here perhaps?
+  if (
```

Comment:
We could probably use buttons in Chatfuel for this rather than accepting freeform input.